thorlcr/thorutil/thormisc.hpp

```diff
@@ -356,9 +356,14 @@ class graph_decl CFileOwner : public CSimpleInterface, implements IInterface
     }
     void noteSize(offset_t size)
     {
-        fileSize = size;
-        if (fileSizeTracker)
-            fileSizeTracker->growSize(fileSize);
+        if (fileSizeTracker && fileSize!=size)
+        {
+            if (size > fileSize)
+                fileSizeTracker->growSize(size-fileSize);
+            else
+                fileSizeTracker->shrinkSize(fileSize-size);
+            fileSize = size;
```

Comment:
If fileSizeTracker is not set, fileSize won't be updated. I think should be : 
```
    if (fileSize!=size)
    {
        if (fileSizeTracker)
        {
            if (size > fileSize)
                fileSizeTracker->growSize(size-fileSize);
            else
                fileSizeTracker->shrinkSize(fileSize-size);
        }
        fileSize = size;
    }
```


( Gavin made a similar comment in this PR : https://github.com/hpcc-systems/HPCC-Platform/pull/18573#discussion_r1637580363 )